src/client.rs

```diff
@@ -45,11 +53,54 @@ impl Client {
             _ => Channel::balance_list(endpoints.into_iter()),
         };
 
-        let kv = KvClient::new(channel.clone(), None);
-        let watch = WatchClient::new(channel.clone(), None);
-        let lease = LeaseClient::new(channel, None);
+        let mut interceptor: Option<Interceptor> = None;
+        if let Some(connect_options) = options {
+            if let Some(user) = connect_options.user {
+                // specify the user's name and password
+                let name = user.0;
+                let password = user.1;
+                let mut tmp_auth = AuthClient::new(channel.clone(), None);
+                let resp = tmp_auth.authenticate(name, password).await?;
+                let token = resp.token().clone();
+
+                interceptor = Some(Interceptor::new(move |mut request| {
+                    let metadata = request.metadata_mut();
+                    // authorization for http::header::AUTHORIZATION
+                    metadata.insert("authorization", token.parse().unwrap());
+                    Ok(request)
+                }));
+            }
+        }
 
-        Ok(Self { kv, watch, lease })
+        if interceptor.is_none() {
+            let kv = KvClient::new(channel.clone(), None);
+            let watch = WatchClient::new(channel.clone(), None);
+            let lease = LeaseClient::new(channel.clone(), None);
+            let lock = LockClient::new(channel.clone(), None);
+            let auth = AuthClient::new(channel, None);
+
+            Ok(Self {
+                kv,
+                watch,
+                lease,
+                lock,
+                auth,
+            })
+        } else {
+            let kv = KvClient::new(channel.clone(), interceptor.clone());
+            let watch = WatchClient::new(channel.clone(), interceptor.clone());
+            let lease = LeaseClient::new(channel.clone(), interceptor.clone());
+            let lock = LockClient::new(channel.clone(), interceptor.clone());
+            let auth = AuthClient::new(channel, interceptor.clone());
+
+            Ok(Self {
+                kv,
+                watch,
+                lease,
+                lock,
+                auth,
+            })
+        }
```

Comment:
The code is redundant, we can simplify it. You can clone `Option` directly.

Reply:
OK